tests/pytests/test.py

```diff
@@ -1993,29 +1993,72 @@ def testInfoCommandImplied(env):
-            env.cmd('ft.del', 'idx', 'doc')
+            conn.execute_command('ft.del', 'idx', 'doc')
         except redis.ResponseError:
             pass
 
-        # Insert a document
-        env.assertCmdOk('ft.add', 'idx', 'doc', 1.0, 'fields',
-                         'body', "located",
-                         'name', "located")
+        # Insert documents
+        conn.execute_command('HSET', 'doc1', 'body', "located", 'name', "located")
+        conn.execute_command('HSET', 'doc2', 'body', "smith", 'name', "smith")
+        conn.execute_command('HSET', 'doc3', 'body', "smiths", 'name', "smiths")
+        conn.execute_command('HSET', 'doc4', 'body', "cherry")
+        conn.execute_command('HSET', 'doc5', 'body', "cherries")
+        conn.execute_command('HSET', 'doc6', 'name', "candy")
+        conn.execute_command('HSET', 'doc7', 'name', "candies")
 
         # Now search for the fields
-        res_body = env.cmd('ft.search', 'idx', '@body:location')
-        res_name = env.cmd('ft.search', 'idx', '@name:location')
-        env.assertEqual(0, res_name[0])
+        res_body = conn.execute_command('ft.search', 'idx', '@body:location')
         env.assertEqual(1, res_body[0])
+        res_name = conn.execute_command('ft.search', 'idx', '@name:location')
+        env.assertEqual(0, res_name[0])
+
+        res_body = conn.execute_command('ft.search', 'idx', '@body:smith')
+        env.assertEqual(2, res_body[0])
+        res_name = conn.execute_command('ft.search', 'idx', '@name:smith')
+        env.assertEqual(1, res_name[0])
+
+        res_body = conn.execute_command('ft.search', 'idx', '@body:smiths')
+        env.assertEqual(2, res_body[0])
+        res_name = conn.execute_command('ft.search', 'idx', '@name:smiths')
+        env.assertEqual(1, res_name[0])
+
+        # Test modifier list
+        # 2 results are returned because 'body' field is stemming 'cherry'
+        res = conn.execute_command('ft.search', 'idx', '@body|name:cherry')
+        env.assertEqual(2, res[0])
+        res = conn.execute_command('ft.search', 'idx', '@body|name:cherries')
+        env.assertEqual(2, res[0])
+
+        # only 1 result is returned because 'name' field is not stemming
+        res = conn.execute_command('ft.search', 'idx', '@body|name:candy')
+        env.assertEqual(1, res[0])
+        res = conn.execute_command('ft.search', 'idx', '@body|name:candies')
+        env.assertEqual(1, res[0])
+
+        # 3 results are returned because 'body' field is stemming 'candy' 
+        # but 'name' field is not stemming  
+        res = conn.execute_command('ft.search', 'idx', '@body|name:candy|cherry')
+        env.assertEqual(3, res[0])
+        res = conn.execute_command('ft.search', 'idx', '@body|name:candies|cherries')
+        env.assertEqual(3, res[0])
+
+        # Test explaincli with modifier list
+        env.expect('ft.explain', 'idx', '@body|name:candy').equal(r'''
+@body|name:UNION {
+  @body|name:candy
+  @body|name:+candi(expanded)
```

Comment:
'name' field is not stemming and is OR'd with `body` field, which is stemming,
So would match `name` using a stem?
Are we taking here the approach that if one or more of the OR'd fields are stemming, then all fields are actually stemming?

Reply:
> So would match name using a stem?

No.
 
>Are we taking here the approach that if one or more of the OR'd fields are stemming, then all fields are actually stemming?

No. We only get stemming results from fields which actually are stemming. In the example: `body` returns 2 results due the the stemming, but `name` only returns 1.
The following queries are equivalent:
`ft.search idx '@body|name:(candy|cherries)' dialect 2`
`ft.search idx '@body:(candy|cherries) | @name:(candy|cherries)' dialect 2`

Reply:
So only `FT.EXPLAIN` is misleading by showing this:
```@body|name:+candi(expanded)```

It looks like the stem is used for both `body` and `name` although when actually querying, only `body` is matched against the stem.
Should it show:
```@body:+candi(expanded)```

Reply:
OK. `StemmerExpander()` function was modified, the field mask of the expanded nodes now reflects only the expandable fields.
```sh
127.0.0.1:6379> ft.explaincli idx '@body|name:candy' 
1) @body|name:UNION {
2)   @body|name:candy
3)   @body:+candi(expanded)
4)   @body:candi(expanded)
5) }
```